src/main/java/duke/Task.java

```diff
@@ -0,0 +1,52 @@
+package duke;
+
+/**
+ * Represents a task to be done.
+ * Is abstract since tasks can be either deadlines, todos or events.
+ */
+public abstract class Task {
+    private final String description;
+    private boolean isCompleted;
+
+    /**
+     * Constructor for a task.
+     * @param description Description of the task
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isCompleted = false;
+    }
+    /**
+     * Constructor for a task.
+     * @param description Description of the task
+     * @param isCompleted Whether the task is completed
+     */
+    public Task(String description, boolean isCompleted) {
+        this.description = description;
+        this.isCompleted = isCompleted;
+    }
+
+    /**
+     * Getter for the description of the task
+     * @return description of the task
+     */
+    public String getDescription() {
+        return this.description;
+    }
+
+    /**
+     * Getter for whether the task is completed
+     * @return Completion status of the task
+     */
+    public boolean getCompletion() {
+        return this.isCompleted;
+    }
+    public void setCompletion() {
```

Comment:
Perhaps naming this method as `toggleCompletion()` would be better?

Reply:
I am mostly sure that the coding standard requires adherence to `get` + the name of the variable for boolean setter methods. Thank you for the other comments though